assets/scripts/info_bubble/WidthControl.jsx

```diff
@@ -0,0 +1,302 @@
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { changeSegmentWidth } from '../store/actions/street'
+import { t } from '../app/locale'
+import { trackEvent } from '../app/event_tracking'
+import { KEYS } from '../app/keyboard_commands'
+
+import { loseAnyFocus } from '../util/focus'
+import {
+  MIN_SEGMENT_WIDTH,
+  MAX_SEGMENT_WIDTH,
+  RESIZE_TYPE_TYPING,
+  resizeSegment,
+  incrementSegmentWidth,
+  scheduleControlsFadeout
+} from '../segments/resizing'
+import {
+  prettifyWidth,
+  undecorateWidth,
+  processWidthInput
+} from '../util/width_units'
+
+const WIDTH_EDIT_INPUT_DELAY = 200
+
+class WidthControl extends React.Component {
+  static propTypes = {
+    touch: PropTypes.bool,
+    segment: PropTypes.object, // eslint-disable-line react/no-unused-prop-types
+    segmentEl: PropTypes.object, // TODO: this is the actual DOM element; only here for legacy reasons
+    position: PropTypes.number,
+    value: PropTypes.number,
+    changeSegmentWidth: PropTypes.func
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.oldValue = null
+    this.timerId = -1
+    this.inputEl = null
+
+    this.state = {
+      isEditing: false,
+      displayValue: prettifyWidth(props.value)
+    }
+  }
+
+  /**
+   * If UI is not in user-editing mode, update the display
+   * when the value in store changes
+   *
+   * @param {Object} nextProps
+   */
+  componentWillReceiveProps (nextProps) {
+    if (!this.state.isEditing) {
+      this.setState({
+        displayValue: prettifyWidth(nextProps.value)
+      })
+    }
+  }
+
+  /**
+   * If UI is going to enter user-editing mode, immediately
+   * save the previous value in case editing is cancelled
+   *
+   * @param {Object} nextProps
+   * @param {Object} nextState
+   */
+  componentWillUpdate (nextProps, nextState) {
+    if (!this.state.isEditing && nextState.isEditing) {
+      this.oldValue = this.props.value
+    }
+  }
+
+  onClickIncrement = (event) => {
+    const segmentEl = this.props.segmentEl
+    const precise = event.shiftKey
+
+    // Legacy: normalize value and update DOM, then return the normalized value
+    const newWidth = incrementSegmentWidth(segmentEl, true, precise, this.props.value)
+
+    // Set new value in Redux
+    this.props.changeSegmentWidth(this.props.position, newWidth)
+
+    scheduleControlsFadeout(segmentEl)
+
+    trackEvent('INTERACTION', 'CHANGE_WIDTH', 'INCREMENT_BUTTON', null, true)
+  }
+
+  onClickDecrement = (event) => {
+    const segmentEl = this.props.segmentEl
+    const precise = event.shiftKey
+
+    // Legacy: normalize value and update DOM, then return the normalized value
+    const newWidth = incrementSegmentWidth(segmentEl, false, precise, this.props.value)
+
+    // Set new value in Redux
+    this.props.changeSegmentWidth(this.props.position, newWidth)
+
+    scheduleControlsFadeout(segmentEl)
+
+    trackEvent('INTERACTION', 'CHANGE_WIDTH', 'INCREMENT_BUTTON', null, true)
```

Comment:
I think we want to track a different event here, right?

Reply:
Yep!